systemtest/benchtest/expvar.go

```diff
@@ -72,12 +76,54 @@ func queryExpvar(out *expvar, srv string) error {
 	}
 	req.Header.Set("Accept", "application/json")
 
+	agg := make(map[string]expvar)
+	var seen int
+	const timesSeen = 10
+	for {
+		var tmp expvar
+		// NOTE(marclop) we could also aggregate based on the beats ephemeral id.
+		// However, that has the drawback of not being to replace a node when it
+		// restarts.
+		// For example, if we push too hard and it falls out of the LB, the beats
+		// ID will change if the APM Server process is restarted, resulting in
+		// some stats never being updated.
+		id, err := doExpvar(req, &tmp)
+		if err != nil {
+			return err
+		}
+		if _, ok := agg[id]; ok {
+			seen++
+		}
+		agg[id] = tmp
+		// We must ensure that we have made enough requests to the remote APM
+		// Server to guarantee with a degree of certainty that all the remote
+		// APM Servers metrics have been queried.
```

Comment:
I suppose that may be slightly more deterministic when it comes to knowing which servers accepted batches. I suppose the best course would be to store these on a per-benchmark basis?

Reply:
I was thinking about it but I don't think your suggestions will work quite well since before we run any benchmark, we need to aggregate the initial stats from the expvar endpoint, which at that point (first benchmark) we haven't sent any data to the APM Server. I'm not sure how we'd do what you suggest given this constraint.

Also, I checked with the cloud proxy team, there isn't a way (currently) to ask the the proxy to route the request to a specific instance, although it would be possible to add it should we require it.

Reply:
@marclop ah good point, thanks for humouring me anyway :)

We might eventually need an alternative way to measure throughput, like querying Elasticsearch. We did that in hey-apm, and I moved away from it to (a) simplify things and (b) enable benchmarking when not using the Elasticsearch output. If things are going to get more complicated by only relying on expvar, though, then perhaps we should revert.